### PR TITLE
APNs integration

### PR DESCRIPTION
**Pull request by raikasdev**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Apple Push Notification service integration.

- **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**
Added the iOS push notification support

- **Other information**:

**NOTICE!** This hasn't been tested yet! I learned that I would need a Apple device to test this, which I don't have.

### CODE REVIEW

File: apps/api/src/app/events/services/push-service/handlers/apns.handler.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { BasePushHandler } from './base.handler';
+import { ICredentials } from '@novu/dal';
+
+export class APNSHandler extends BasePushHandler {
+  constructor() {
+    super('apns', ChannelTypeEnum.PUSH);
+  }
+
+  buildProvider(credentials: ICredentials) {
+    console.log('apns test');
```
Comment (davidsoderberg) [inline apps/api/src/app/events/services/push-service/handlers/apns.handler.ts:12]: Console.log ...

File: apps/api/src/app/events/services/push-service/push.factory.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 export class PushFactory implements IPushFactory {
-  handlers: IPushHandler[] = [new FCMHandler()];
+  handlers: IPushHandler[] = [new FCMHandler(), new APNSHandler()];
 
   getHandler(integration: IntegrationEntity): IPushHandler {
     try {
       const handler =
         this.handlers.find((handlerItem) => handlerItem.canHandle(integration.providerId, integration.channel)) ?? null;
-
+      console.log(handler);
```
Comment (davidsoderberg) [inline apps/api/src/app/events/services/push-service/push.factory.ts:13]: Same here ;)

File: packages/node/src/lib/subscribers/subscriber.interface.ts [hunk 3]
```diff
@@ -92,6 +94,58 @@ export type ITriggerOverrideFCM = {
...
+    | {
+        title?: string;
+        subtitle?: string;
+        body: string;
+        'title-loc-key'?: string;
+        'title-loc-args'?: string[];
+        'action-loc-key'?: string;
+        'loc-key'?: string;
+        'loc-args'?: string[];
+        'launch-image'?: string;
```
Comment (samsoft00) [inline packages/node/src/lib/subscribers/subscriber.interface.ts:142]: We can extract this to separate interface

File: apps/api/src/app/events/services/push-service/handlers/apns.handler.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { ICredentials } from '@novu/dal';
+
+export class APNSHandler extends BasePushHandler {
+  constructor() {
+    super('apns', ChannelTypeEnum.PUSH);
+  }
+
+  buildProvider(credentials: ICredentials) {
+    if (!credentials.secretKey || !credentials.apiKey || !credentials.projectName) {
+      throw Error('Config is not valid for apns');
```
Comment (k88manish) [inline apps/api/src/app/events/services/push-service/handlers/apns.handler.ts:13]: Shouldn't this be `throw new Error(....)`

**Review by k88manish — COMMENTED** [review]
some comment

**Comment by scopsy** [comment]
@raikasdev GitHub automatically closed this PR :((( could you reopen this against main please?

**Pull request closed without merging**